knimin/handlers/barcode_util.py

```diff
@@ -172,8 +202,61 @@ def _build_email(self, login_user, barcode, email_type,
         return subject, body_message
 
 
+def align_with_qiita_categories(samples, categories):
+    # obviously should instead align to the actual sample metadata...
+    aligned = {}
+    for s in samples:
+        aligned[s] = {c: 'LabControl test' for c in categories}
+    return aligned
+
+
+@set_access(['Scan Barcodes'])
+class PushQiitaHandler(BaseHandler):
+    executor = concurrent.futures.ThreadPoolExecutor(5)
+    study_id = config.qiita_study_id
+    qclient = get_qiita_client()
+
+    @concurrent.run_on_executor
+    def _push_to_qiita(self, study_id, samples):
+        # TODO: add a mutex or block to ensure a single call process at a time
+        cats = self.qclient.get('/api/v1/study/%s/samples/info' % study_id)
```

Comment:
Should `/api/v1/study/` be a class variable in case it changes?

Reply:
It also obfuscates the endpoint. These haven't changed in 2 years and I don't believe there is a plan for a v2?

Reply:
Not sure ... let's leave as is.